### PR TITLE
fix(cli): invalid plugin template

### PR DESCRIPTION
**Pull request by lucasfernog**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: tooling/cli/templates/plugin/__example-api/tauri-app/vite.config.js [hunk 1]
```diff
@@ -1,27 +1,45 @@
...
+    // https://tauri.studio/v1/api/config#buildconfig.beforedevcommand
+    envPrefix: ["VITE_", "TAURI_ENV_"],
+    build: {
+      // Tauri supports es2021
+      target: ["es2021", "chrome100", "safari13"],
+      // don't minify for debug builds
+      minify: !process.env.TAURI_ENV_DEBUG ? "esbuild" : false,
+      // produce sourcemaps for debug builds
+      sourcemap: !!process.env.TAURI_ENV_DEBUG,
+    },
```
Comment (amrbashir) [inline tooling/cli/templates/plugin/__example-api/tauri-app/vite.config.js]: let's remove all of this, not needed

File: tooling/cli/templates/plugin/__example-api/tauri-app/vite.config.js [hunk 2]
```diff
@@ -1,27 +1,45 @@
...
-  plugins: [svelte()],
+export default defineConfig(async ({
+  command,
+  mode
+}) => {
+  const host =
+    process.env.TAURI_ENV_PLATFORM === 'android' ||
+    process.env.TAURI_ENV_PLATFORM === 'ios' ?
+    await internalIpV4() :
+    'localhost'
```
Comment (amrbashir) [inline tooling/cli/templates/plugin/__example-api/tauri-app/vite.config.js]:
```suggestion
  const mobile = !!/android|ios/.exec(process.env.TAURI_ENV_PLATFORM);
```

File: tooling/cli/templates/plugin/__example-api/tauri-app/vite.config.js [hunk 3]
```diff
@@ -1,27 +1,45 @@
...
+    // tauri expects a fixed port, fail if that port is not available
+    server: {
+      host: '0.0.0.0',
+      port: 1420,
+      strictPort: true,
+      hmr: {
+        protocol: 'ws',
+        host,
+        port: 1420
+      },
```
Comment (amrbashir) [inline tooling/cli/templates/plugin/__example-api/tauri-app/vite.config.js]:
```suggestion
    server: {
    port: 1420,
    strictPort: true,
    host: mobile ? "0.0.0.0" : false,
    hmr: mobile
      ? {
          protocol: "ws",
          host: await internalIpV4(),
          port: 1421,
        }
      : undefined,
  },	
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Comment by FabianLars** [comment]
i think `internal-ip` is missing from the termplate's package.json

**Comment by lucasfernog** [comment]
I wonder how the hell it worked on my demo plugin test.

**Review by amrbashir — APPROVED** [review]

**Pull request merged**